examples/signal/plot_covariance_estimation.py

```diff
@@ -0,0 +1,208 @@
+# Matrix condition number is the ratio between the highest and lowest
+# eigenvalues: high values indicates ill-conditionned matrices that are not
+# suitable for classification.
+# A common approach to mitigate this issue is to regularize covariance matrices
+# by shrinkage, like in Ledoit-Wolf, Schaefer-Strimmer or oracle estimator.
+
+estimators = ["lwf", "oas", "sch", "scm"]
+tmin = -0.2
+w_len = np.linspace(0.2, 2, 10)
+n_trials = 45
+dfc = list()
+
+for wl in w_len:
+    epochs = Epochs(
+        raw,
+        events,
+        event_ids,
+        tmin,
+        tmin + wl,
+        picks=picks,
+        preload=True,
+        verbose=False,
+    )
+    for est in estimators:
+        cov = Covariances(estimator=est).transform(epochs.get_data())
+        for k in range(len(cov)):
+            ev, _ = np.linalg.eigh(cov[k, :, :])
+            dfc.append(dict(estimator=est, wlen=wl, cond=ev[-1] / ev[0]))
+dfc = pd.DataFrame(dfc)
+
+###############################################################################
+
+fig, ax = plt.subplots(figsize=(6, 4))
+ax.set(yscale="log")
+sns.lineplot(data=dfc, x="wlen", y="cond", hue="estimator", ax=ax)
+ax.set_title("Condition number of estimated covariance matrices")
+ax.set_xlabel("Epoch length (s)")
+ax.set_ylabel(r"$\lambda_{\min}$/$\lambda_{\max}$")
+plt.tight_layout()
+
+###############################################################################
+# Picking a good estimator for classification
+# -----------------------------------------------
+# The choice of covariance estimator have an impact on classification,
+# especially when the covariances are estimated on short time windows.
+
+estimators = ["lwf", "oas", "sch", "scm"]
+tmin = 0.0
+w_len = np.linspace(0.2, 2.0, 5)
+n_trials, n_splits = 45, 3
+dfa = list()
+sc = "balanced_accuracy"
+
+cv = StratifiedKFold(n_splits=n_splits, shuffle=True)
+for wl in w_len:
+    epochs = Epochs(
+        raw,
+        events,
+        event_ids,
+        tmin,
+        tmin + wl,
+        proj=True,
+        picks=picks,
+        preload=True,
+        baseline=None,
+        verbose=False,
+    )
+    X = epochs.get_data()
+    y = np.array([0 if ev == 2 else 1 for ev in epochs.events[:, -1]])
+    for est in estimators:
+        clf = make_pipeline(Covariances(estimator=est), MDM())
+        try:
+            score = cross_val_score(clf, X, y, cv=cv, scoring=sc)
+            dfa += [dict(estimator=est, wlen=wl, accuracy=sc) for sc in score]
+        except ValueError:
+            print(f"{est}: {wl} is not sufficent to estimate a PSD matrix")
+            dfa += [dict(estimator=est, wlen=wl, accuracy=np.nan)] * n_splits
+dfa = pd.DataFrame(dfa)
+
+###############################################################################
+
+fig, ax = plt.subplots(figsize=(6, 4))
+sns.lineplot(
+    data=dfa,
+    x="wlen",
+    y="accuracy",
+    hue="estimator",
+    style="estimator",
+    ax=ax,
+    ci=None,
```

Comment:
Why do not display the accuracy variability with ci='sd'?

Reply:
It was a compromise, because they are quite large as I tried to keep computation low for generating docs.

Reply:
Previous lineplots use default value, ie `ci=95`, and it does not increase computation time for doc generation.

Reply:
Yes, on previous artificial examples `ci=95` is OK. In this part, we download EEG data from physionet. If we plot the confidence interval here, it will be quite large and make lineplots difficult to read. To tighten the confidence interval, we should use more trials and, ideally, more subject (just one here), that will significatively increase the doc generation time.